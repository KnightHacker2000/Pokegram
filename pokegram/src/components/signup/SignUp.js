/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable no-console */
/* eslint-disable no-trailing-spaces */
import React, { useState } from 'react';
import Avatar from '@mui/material/Avatar';
import Button from '@mui/material/Button';
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import Divider from '@mui/material/Divider';
import Typography from '@mui/material/Typography';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { TextField } from '@mui/material';
import validator from 'validator';
import pokemon from '../../images/pikachu.jpg';
import Upload from '../posts/Upload';
import HomeState from '../../models/homeState';

const theme = createTheme();

function SignUp(props) {
  const [user, setUser] = useState({
    fullname: '',
    username: '',
    emailaddress: '',
    password: ''
  });
  const handleSubmit = (event) => {
    event.preventDefault();
    // alert('current state is: ' + user.username + ' ' + user.password);
    
    // testing code for state manipulation
    console.log(props);
    props.parStates.handleSetStates(true, true, user.username);

    // TODO: create new user, call register endpoint
  };
  const handleFullname = (event) => {
    console.log(event.target.value);
    const updateuser = {
      fullname: event.target.value,
      username: user.username,
      emailaddress: user.emailaddress,
      password: user.password
    };
    setUser(updateuser);
  };
  const handleUsername = (event) => {
    console.log(event.target.value);
    const updateuser = {
      fullname: user.fullname,
      username: event.target.value,
      emailaddress: user.emailaddress,
      password: user.password
    };
    setUser(updateuser);
  };
  const handleEmailaddress = (event) => {
    console.log(event.target.value);
    const updateuser = {
      ullname: user.fullname,
      username: user.username,
      emailaddress: event.target.value,
      password: user.password
    };
    setUser(updateuser);
  };
  const handlePassword = (event) => {
    console.log(event.target.value);
    const updateuser = {
      fullname: user.fullname,
      username: user.username,
      emailaddress: user.emailaddress,
      password: event.target.value
    };
    setUser(updateuser);
  };
  const handleSignin = (event) => {
    event.preventDefault();
    props.parStates.handleSetStates(false, true, user.username);
  };
  return (
    <ThemeProvider theme={theme}>
      <Container maxWidth="xs">
        <Box sx={{
          marginTop: 8,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center'
        }}
        >
          <Avatar alt="pikachu" sx={{ m: 1 }} src={pokemon} />
          <Typography component="h1" variant="h4">Wecome to Pokegram!</Typography>
          <Typography component="h2" variant="h6">Register with Pokemongram, Pika Pika</Typography>
        </Box>
        <Divider variant="middle" />
        <Box 
          component="form" 
          onSubmit={handleSubmit} 
          noValidate 
          sx={{
            marginTop: 20, my: 3, mx: 2, p: 4, border: '2px solid', borderRadius: '16px', boxShadow: 1, alignItems: 'center'
          }}
        >
          <Typography component="h1" variant="h6">Full Name:</Typography>
          <TextField 
            margin="normal" 
            required 
            fullWidth 
            id="fullname" 
            label="Fullname"
            name="fullname"
            value={user.fullname}
            onChange={handleFullname} 
          />
          <Typography component="h1" variant="h6">Username:</Typography>
          <TextField margin="normal" required fullWidth id="username" label="Username" onChange={handleUsername} />
          <Typography component="h1" variant="h6">Password:</Typography>
          <TextField 
            margin="normal" 
            type="password"
            label="Password"
            required 
            fullWidth 
            id="password" 
            name="password"
            value={user.password}
            onChange={handlePassword} 
            error={user.password !== '' && user.password.length < 8}
            helperText={user.password !== '' && user.password.length < 8 ? 'Password should contains Minimum eight characters' : ' '}
          />
          <Typography component="h1" variant="h6">Email Address:</Typography>
          <TextField 
            margin="normal" 
            required 
            fullWidth 
            id="emailaddress" 
            label="Email address"
            name="emailaddress"
            value={user.emailaddress}
            onChange={handleEmailaddress}
            error={user.emailaddress !== '' && !validator.isEmail(user.emailaddress)}
            helperText={user.emailaddress !== '' && !validator.isEmail(user.emailaddress) ? 'invalid email address' : ' '} 
          />
          <Button type="submit" fullWidth variant="contained" sx={{ mat: 3, mb: 2 }} onClick={handleSignin}>Sign Up</Button>
        </Box>
        <Box sx={{ 
          my: 3, mx: 2, p: 1, marginTop: 8, border: 1, alignItems: 'center', display: 'flex'
        }}
        >
          <Typography component="h1" variant="h6" style={{ marginLeft: '16px', marginRight: '64px' }}>Had an Account?</Typography>
          <Button type="submit" variant="contained" sx={{ marginTop: '16px', mat: 3, mb: 2 }} onClick={handleSignin}>Sign In</Button>
        </Box>
      </Container>
    </ThemeProvider>
  );
}
export default SignUp;
