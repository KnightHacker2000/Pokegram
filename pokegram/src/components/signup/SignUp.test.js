/**
 * @jest-environment jsdom
 */
 import React from 'react';
 import { render, screen, fireEvent } from '@testing-library/react';
 import renderer from 'react-test-renderer';
 import SignUp from './SignUp';
 import RootState from '../../models/rootState';
 import '@testing-library/jest-dom/extend-expect';
 

const parentStates = new RootState(
  false,
  false,
  -1,
  (newLogin, newReg, newMyUID) => {
    console.log(newLogin, newReg, newMyUID);
  }
);

 test('test SignUp snapshot', () => {
   const SignUpComp = renderer.create(<SignUp />);
   const tree = SignUpComp.toJSON();
   // console.log(SignUpComp.toJSON());
   expect(tree).toMatchSnapshot();
 });
 
 // test('test button click')
 test('SignUp button', () => {
   render(<SignUp parStates={parentStates}/>);
  const submit = screen.getByTestId("signUp_submit");
  // const password = screen.getByTestId("test_username");
  // expect(password.value).toBe(undefined);
  console.log(submit);
  fireEvent.click(submit);
 });

 test('SignIn button', () => {
    render(<SignUp parStates={parentStates} />);
    const submit = screen.getByTestId("signIn_submit");
    fireEvent.click(submit);
    // console.log(signInsubmit);
  });