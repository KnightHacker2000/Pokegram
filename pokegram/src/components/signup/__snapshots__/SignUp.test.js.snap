// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SignUp snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
>
  <div
    className="MuiBox-root css-binzgt"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular css-ft7g56-MuiAvatar-root"
    >
      <img
        alt="pikachu"
        className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="http://img4.wikia.nocookie.net/__cb20140328190757/pokemon/images/thumb/2/21/001Bulbasaur.png/200px-001Bulbasaur.png"
      />
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Wecome to Pokegram!
    </h1>
    <h2
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Register with Pokemongram, Pika Pika
    </h2>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
  />
  <form
    className="MuiBox-root css-1wk4014"
    noValidate={true}
    onSubmit={[Function]}
  >
    <h1
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Full Name:
    </h1>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="fullname"
        id="fullname-label"
      >
        Fullname
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="fullname"
          name="fullname"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Fullname
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Username:
    </h1>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="username"
        id="username-label"
      >
        Username
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Username
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Password:
    </h1>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="password"
        id="password-label"
      >
        Password
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-describedby="password-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="password"
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="password"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Password
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-required css-1wc848c-MuiFormHelperText-root"
        id="password-helper-text"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </p>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Email Address:
    </h1>
    <div
      className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor="emailaddress"
        id="emailaddress-label"
      >
        Email address
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-describedby="emailaddress-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id="emailaddress"
          name="emailaddress"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Email address
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-required css-1wc848c-MuiFormHelperText-root"
        id="emailaddress-helper-text"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-axev6t-MuiButtonBase-root-MuiButton-root"
      data-testid="signUp_submit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <div
    className="MuiBox-root css-3i7hrm"
  >
    <h1
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      style={
        {
          "marginLeft": "16px",
          "marginRight": "64px",
        }
      }
    >
      Had an Account?
    </h1>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1h553qu-MuiButtonBase-root-MuiButton-root"
      data-testid="signIn_submit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Sign In
    </button>
  </div>
</div>
`;
